pkg/validation/validation.go

```diff
@@ -114,6 +121,35 @@ func nodeGroupsDuplicates(nodeGroups []nropv1.NodeGroup) error {
 	return nil
 }
 
+// TODO: move it under the validation webhook once we will have one
+func nodeGroupsDuplicatesByPoolName(nodeGroups []nropv1.NodeGroup) error {
+	duplicates := map[string]int{}
+	for _, nodeGroup := range nodeGroups {
+		if nodeGroup.PoolName == nil {
+			continue
```

Comment:
this must never happen so it's worth a log (`klog.InfoS`)

Reply:
Thanks for the comment; depending on the direction we went with, the user should set either MCPselector or PoolName, so one of them should be nil. We do not map MCPSelector to PoolName (and we can't at the moment because matching mcps can be > one).
This function is checking the NodeGroupSpec where PoolName is a pointer. however on NodeGroupStatus PoolName must be set (derived from the trees). can you elaborate more please on what I'm missing for this condition to never be true?

Reply:
Right, we must still support identification by MCPSelector. Hence the code looks correct as it is